app/db/models/index.js

```diff
@@ -374,26 +375,33 @@ module.exports = function (sequelize, env) {
         .error(callback);
     },
 
-    createResetCode: function (user, appURL, callback) {
+    createResetCode: function (user, appURL, oauthParams, callback) {
+
       var code = hat(RESET_CODE_BIT_LENGTH, RESET_CODE_BASE);
       var userResetCode = resetCode.build({
         code: code,
         UserId: user.id
       });
 
+      var resetUrlObj = url.parse(appURL || env.get("WEBMAKERORG"), true);
```

Comment:
Couldn't appURL just contain the querystring parameters needed, instead of having to pass them separately?